### PR TITLE
ref(snuba): add typing to tsdb/snuba and remove from ignore list

### PR DESCRIPTION
**Pull request by kylemumma**
This pr is associated with [SNS-2588](https://getsentry.atlassian.net/browse/SNS-2588)

The goal of this pr is to fully type utils/snuba.py up to the standard deemed in the codebase's pyproject.toml, and remove it from the do-not-typecheck list.

[SNS-2588]: https://getsentry.atlassian.net/browse/SNS-2588?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/sentry/utils/snuba.py
Comment (kylemumma) [inline src/sentry/utils/snuba.py:1]: this was black extension

Comment (kylemumma) [inline src/sentry/utils/snuba.py:1]: this was my black plugin

File: src/sentry/tsdb/base.py [hunk 1]
```diff
@@ -212,7 +212,7 @@ def normalize_ts_to_rollup(self, epoch, seconds):
         """
         return int(epoch / seconds)
 
-    def get_optimal_rollup(self, start_timestamp, end_timestamp):
+    def get_optimal_rollup(self, start_timestamp, end_timestamp) -> int:
```
Comment (wedamija) [inline src/sentry/tsdb/base.py:215]: Looks like we're still missing types for a fair few input params here and on other functions

Comment (kylemumma) [reply src/sentry/tsdb/base.py:215]: @wedamija Thank you for the review Dan. I was upgrading the typing for this file based on the ```# minimal strictness settings``` of the mypy configuration in pyproject.toml

I noticed at the bottom of pyproject.toml there seemed to be an override with ```disallow_untyped_defs``` for a few modules. Is that what I should be following?

Comment (wedamija) [reply src/sentry/tsdb/base.py:215]: I'm not totally sure tbh, but it seems to me that we should try and fully type modules when possible. #discuss-typing might have more info

Comment (evanh) [reply src/sentry/tsdb/base.py:215]: The current methodology is to type things to the point that the Sentry type checker is passing. Then we can slowly ramp up the strictness.

Normally I would agree with Dan about fully typing this module, but it's a big one, and legacy, and used in a lot of places, so getting it to pass the current strictness of typing feels correct to me. We can tackle the typing in smaller chunks.

**Review by evanh — APPROVED** [review]

**Pull request merged**